libsolidity/interface/CompilerStack.cpp

```diff
@@ -150,10 +149,12 @@ bool CompilerStack::parse()
 			}
 		}
 	}
-	if (!Error::containsOnlyWarnings(m_errors))
-		// errors while parsing. should stop before type checking
-		return false;
 
+	return Error::containsOnlyWarnings(m_errors);
```

Comment:
I guess we should clear `m_SourceUnitsByName()` if this fails if we want to keep 100% compatibility with the previous case.